packages/api/src/webhooks/cannon.ts

```diff
@@ -202,7 +210,7 @@ export default class WebhookCannon {
   }
 
   async onTrigger(event: Queue) {
```

Comment:
Another thing that came to my mind about retries:

We currently use a single event for all webhooks, fetching the webhook list when consuming that event. I think that's nice, and it make the firing of the events/webhooks really cheap, only needing to send 1 event to rabbit and almost no other I/O.

But I think there might be some issues in case we have multiple webhooks registered for the same event. Say, one global webhook and one specific to a streamId, or maybe one that listen to all events (someday?) and one that listens to only that specific event. In those cases, if 1 of them fails, we will be retrying the same original `event` object which will trigger all of the webhooks again. Maybe worse, if all of them fail, we will multiply the retries sending 1 retry event for each failure and so the number of events multiplies by `N` on every retry.

One solution that came to my mind would be adding a new field only for the retry events. A `webhookId` field to the event we send to rabbit for retry, and if the event we receive has that field we will fire only that specific hook. This keeps the simple&cheap publishing logic but makes the retry independent for each hook. WDYT?